docs/terminology.md

```diff
@@ -0,0 +1,21 @@
+# AutoRA Terminology
+
+The following table includes naming conventions used throughout AutoRA.
+
+| Term               | Description                                                                                                                                 | Relevant Modules                                |
+|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------|
+| State              | Base object that has all relevant fields and which uses the Delta mechanism to modify those fields.                                         | Core                                            |
```

Comment:
Is there a need to mention the Delta mechanism? It doesn't come up in documentation or tutorials, but that's maybe because we always use the StardardState as far as I've seen.

Reply:
good point. i was going off this: https://docs.google.com/presentation/d/1OdP4M2ByW8GKTFvcCgDG5HDzUwHKdFEhj1ddH4ClvEw/edit#slide=id.g27e19158dea_1_0

which is what we agreed to do in the task: https://github.com/AutoResearch/autora/issues/516#issuecomment-1716300791

Reply:
I think maybe you misunderstood me. I think having the State here is a good idea, but in the description you say "which uses the Delta mechanism" but, unless I am mistaken, we don't talk about the Delta mechanism anywhere. If we want to include it here, we might also want to add it as a description in this table. I am happy with whatever here though.